includes/class-ajax-data-retrieval.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * The core plugin class file
+ *
+ * Defines the functions necessary to register our AJAX functions
+ * in WordPress that will
+ *
+ * @since      0.0.2
+ *
+ * @package    WSUWP_Plugin_iDonate_Ajax_Data
+ * @author     Blair Lierman <blair.lierman@wsu.edu>
+ */
+class WSUWP_Plugin_iDonate_Ajax_Data {
+
+	/**
+	 * Initializes the actions necessary for hooking in our AJAX methods
+	 *
+	 * @since    0.0.2
+	 */
+	public function init() {
+
+		add_action( 'wp_ajax_nopriv_wsuwp_plugin_idonate_ajax_fund_search', array( $this, 'wsuwp_plugin_idonate_ajax_fund_search' ) );
+		add_action( 'wp_ajax_wsuwp_plugin_idonate_ajax_fund_search', array( $this, 'wsuwp_plugin_idonate_ajax_fund_search' ) );
```

Comment:
`wp_ajax` will work for this, but you may want to consider the ease of the WordPress REST API, which is available on all sites on the platform. Post types can be [exposed](https://github.com/washingtonstateuniversity/WSU-25x2030-Theme/blob/master/includes/class-drive-story-post-type.php#L77) in the API and then [consumed in JS](https://github.com/washingtonstateuniversity/WSU-25x2030-Theme/blob/master/js/evidence.js#L93) via that post type's [endpoint](https://github.com/washingtonstateuniversity/WSU-25x2030-Theme/blob/2d4d085cdf81124192b7117516e11e8a2d395fa4/functions.php#L108).

That's not necessarily the way to go, as it requires the post type is public, but may be the easiest as far as handling the results in JS. That said, with the admin ajax approach already implemented, definitely not a requirement.